pkg/logging/logger.go

```diff
@@ -0,0 +1,148 @@
+package logging
+
+import (
+	"github.com/rs/zerolog/pkgerrors"
+	"goharvest2/pkg/conf"
+	"io"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/rs/zerolog"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	defaultLogFileName           string = "harvest.log"
+	defaultLogLevel                     = zerolog.InfoLevel
+	defaultConsoleLoggingEnabled bool   = true
+	defaultFileLoggingEnabled    bool   = true
+	defaultLogMaxMegaBytes       int    = 10 // 10MB
+	defaultLogMaxBackups         int    = 10
+	defaultLogMaxAge             int    = 30
+)
+
+// Configuration for logging
+type LogConfig struct {
+	// Enable console logging
+	ConsoleLoggingEnabled bool
+	// Log Level
+	LogLevel zerolog.Level
+	// Prefix
+	PrefixKey   string
+	PrefixValue string
+	// FileLoggingEnabled makes the framework log to a file
+	FileLoggingEnabled bool
+	// Directory to log to to when filelogging is enabled
+	Directory string
+	// Filename is the name of the logfile which will be placed inside the directory
+	Filename string
+	// MaxSize the max size in MB of the logfile before it's rolled
+	MaxSize int
+	// MaxBackups the max number of rolled files to keep
+	MaxBackups int
+	// MaxAge the max age in days to keep a logfile
+	MaxAge int
+}
+
+var logger *Logger
+
+var once sync.Once
+
+type Logger struct {
+	*zerolog.Logger
+}
+
+func GetInstance() *Logger {
```

Comment:
what about `Get()` instead of `GetInstance()` or `Inst()`

`GetInstance()` is kinda verbose, especially when you consider that the package already communicates logging

Reply:
changed to Get()